### PR TITLE
nanomq: 0.16.3 -> 0.18.2

### PR DESCRIPTION
**Pull request by risicle**
###### Description of changes

Bump. Got tests _mostly_ working, but not enabled by default.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 232475` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nanomq</li>
  </ul>
</details>

**Comment by risicle** [comment]
Is that an approval? :)

File: pkgs/servers/mqtt/nanomq/default.nix [hunk 1]
```diff
@@ -1,25 +1,65 @@
...
     owner = "emqx";
     repo = "nanomq";
     rev = finalAttrs.version;
-    hash = "sha256-9w4afVxuJbYrkagpAe1diftDnjrRjunyhJdJ0BZq3K0=";
+    hash = "sha256-ld+UuuIWcAIt4XkC+wTbgwNUoIyyyjWi+U90YL3I3Tw=";
     fetchSubmodules = true;
   };
 
+  patches = [
+    # https://github.com/emqx/nanomq/pull/1250
```
Comment (sikmir) [inline pkgs/servers/mqtt/nanomq/default.nix:53]: Why not `fetchpatch` https://github.com/emqx/nanomq/pull/1250/commits/ee5fdb37a0f21b238369aef94334f8b8a429647b?

File: pkgs/servers/mqtt/nanomq/default.nix [hunk 2]
```diff
@@ -1,25 +1,65 @@
...
+      hash = "sha256-3DS9DuzHN7BevfgiekUmKKH9ej9wKTrt6Fuh427NC4I=";
+    };
+
+    nativeBuildInputs = [ cmake ninja flex bison ];
+  };
+
+in stdenv.mkDerivation (finalAttrs: {
   pname = "nanomq";
-  version = "0.16.3";
+  version = "0.18.1";
```
Comment (sikmir) [inline pkgs/servers/mqtt/nanomq/default.nix:42]: 0.18.2 is out meanwhile.

File: pkgs/servers/mqtt/nanomq/default.nix (hunk 1 above)
Comment (risicle) [reply pkgs/servers/mqtt/nanomq/default.nix:53]: Ah - I completely forgot to update this after that was merged.

Standby...

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 232475` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nanomq</li>
  </ul>
</details>

**Review by sikmir — APPROVED** [review]

**Pull request merged**